sdks/python/apache_beam/io/gcp/gcsio.py

```diff
@@ -381,12 +381,20 @@ def size_of_files_in_glob(self, pattern):
     file_sizes = {}
     while True:
       response = self.client.objects.List(request)
-      for item in response.items:
+      items = response.items
+      if limit is not None and (len(file_sizes) + len(response.items)) >= limit:
+        items = response.items[:(
+            len(response.items) + len(file_sizes) - limit)]
+      for item in items:
         if fnmatch.fnmatch(item.name, name_pattern):
           file_name = 'gs://%s/%s' % (item.bucket, item.name)
           file_sizes[file_name] = item.size
+      if len(file_sizes) % 10000 == 0:
```

Comment:
Seems like we won't hit this unless total size after a batch end up being a multiple of 10k.

Pls fix this by moving this check and log to for loop above. Also prob we should use a counter instead of calling len(file_sizes) too frequently.